lib/editor/note-content-editor.tsx

```diff
@@ -169,16 +204,18 @@ class NoteContentEditor extends Component<Props> {
       this.state.editor === 'full' &&
       prevProps.searchQuery !== this.props.searchQuery
     ) {
+      this.setState({ selectedSearchMatchIndex: null });
```

Comment:
for state updates it's more appropriate to set them in `getDerivedStateFromProps()` so that it avoids an additional component re-render and the intermediary partial-update

this often requires us to trap more data in local state as `getDerivedStateFromProps()` is intentionally a static class method.

```ts
static getDerivedStateFromProps(props, state) {
	…
	
	return Object.assign( {},
		props.content !== state.content && { content: … },
		…
		props.searchQuery !== state.searchQuery && { selectedSearchMatchIndex: null },
	);
}
```